### ISSUE TITLE
Broadcom ICOS/Fastpath Support?

### ISSUE THREAD

**Issue by alepodj** [networktocode/ntc-templates#726]
##### ISSUE TYPE

 - Template Creation Help

##### Question

- Broadcom ICOS/Fastpath

Are these type of devices/os supported? or any of the current supported os templates compatible with them? 

I see 'linux' is on the os_choices list, and a show version on these types of devices show linux as the OS but i dont see any template for Broadcom ICOS/Fastpath specifically.  

If nothing exists for such, i cant claim to understand how to create new templates(currently going thru it to try and do my best) and i could use some help, but i work with more than 5 or more different models of switches that use ICOS/Fastpath and they are increasing in popularity and im able to provide plenty sample outputs for different commands for that OS type.

As a sample here is a output of the command `show version` as is:

```
(switch1) #show version

Switch: 1

System Description............................. Accton AS4610-54P , 3.2.2.42, Linux 3.6.5, U-Boot 2012.10-gd563f4a (Apr 13 2017 - 12:08:06) - 2012.10.0.5
Machine Type................................... Accton AS4610-54P
Machine Model.................................. BCM-56643
Serial Number.................................. EC1726004357
Maintenance Level.............................. A
Manufacturer................................... 0xbc00
Burned In MAC Address.......................... A8:2B:B5:55:58:01
Software Version............................... 3.2.2.42
Operating System............................... Linux 3.6.5
Network Processing Device...................... BCM56340_A0
Additional Packages............................ BGP-4
                                                QOS
                                                Multicast
                                                IPv6
                                                Routing
                                                Data Center
                                                OpEN API
                                                Prototype Open API


(switch1) #
```

Other known models among others: `Accton 5610 or Quanta LB8 and LB9`

Thanks

**Comment by FragmentedPacket** [comment]
@alepodj After doing some research, it seems like the OS should be `broadcom_icos`, but let us discuss internally and then I'll get back to you with the final answer.

**Comment by alepodj** [comment]
Excellent, yeah do your thing :)

Sounds like a good opportunity to add more to the ntc_templates world.

Any info needed i can provide.

**Comment by FragmentedPacket** [comment]
We're fine with the above: `broadcom_icos` if you want to start adding templates. It will need to be added to `tests/test_index_order.py` and the `os_choices` within that as well for testing purposes.

Please keep the PRs for the templates to a single additional template to help speed up our review process and get each template merged in without others blocking it. More details can be found in our README, but each template requires the accompanying `.raw` and `.yml` files within the test directory for the template.

**Comment by alepodj** [comment]
Great, i already started with a show version command to get familiar with the process. Here is what i have so far:

- Added `broadcom_icos` to `os_choices` under `tests/test_index_order.py`
- Created the template and added it under `templates` folder with name `broadcom_icos_show_version.textfsm`
- Modified the index file in the order defined in the instructions, snippet: 

```
...
avaya_ers_show_vlan.textfsm, .*, avaya_ers, sh[[ow]] vlan
avaya_ers_show_mlt.textfsm, .*, avaya_ers, sh[[ow]] ml[[t]]

avaya_vsp_show_software.textfsm, .*, avaya_vsp, sho[[w]] so[[ftware]]

broadcom_icos_show_version.textfsm, .*, broadcom_icos, sh[[ow]] ver[[sion]]

brocade_fastiron_show_lldp_neighbors_detail.textfsm, .*, brocade_fastiron, sh[[ow]] ll[[dp]] n[[eighbors]] d[[etail]]
brocade_fastiron_show_running-config_vlan.textfsm, .*, brocade_fastiron, sh[[ow]] ru[[nning-config]] v[[lan]]
...
```

- Created a new folder structure under `test` like: `test` -> `broadcom_icos` -> `show version`
- Inside the above folder i have 5 .raw and 5 .yml
   *The .raws as they came from the device output without the command itself
   *The yml generated using `development_scripts.py -c path_to_raw` and the output matches my expected structure

###Current Issue
The tox command fails, im doing it cause its part of the instructions, is it expected int his case?
[tox.txt](https://github.com/networktocode/ntc-templates/files/4697064/tox.txt)

###Context
Im rather new at the whole textfsm still wrapping my head around some of its inners and honestly not an expert at github all together, what else i could be missing?
I created a branch/fork of the original and thats where im tracking all the changes: https://github.com/alepodj/ntc-templates

**Comment by alepodj** [comment]
I had to copy the templates folder manually to: /mnt/c/projects/ntc_templates/ntc-templates/.tox/py36/lib/python3.6/

Tests are going thru now...

**Comment by alepodj** [comment]
All tests passed. I just do a pull request from https://github.com/alepodj/ntc-templates ???

[tox_test_pass_broadcom_icos_show_version.txt](https://github.com/networktocode/ntc-templates/files/4698381/tox_test_pass_broadcom_icos_show_version.txt)

**Comment by FragmentedPacket** [comment]
@alepodj You should be able to navigate to Pull Requests on this repo and it should show that you can open a PR. If not, do the following:

1. Navigate to Pull Request page on our Github repo.
2. Click the `New Pull Request` button.
3. Pick your fork and branch the changes are on, fill in the information (fixes this issue) and then submit it.
4. We'll review it and merge if no changes need to be made.

**Comment by FragmentedPacket** [comment]
This was implemented on #733 and now the repo has the necessary updates to support further templates.
